kiwi/repository/zypper.py

```diff
@@ -263,6 +263,20 @@ def delete_all_repos(self):
         Path.wipe(self.shared_zypper_dir['reposd-dir'])
         Path.create(self.shared_zypper_dir['reposd-dir'])
 
+    def delete_repo_cache(self, name):
+        """
+        Delete zypper repository cache
+
+        The cache data for each repository is stored in a directory
+        of the same name as the repository name. The method deletes
+        this directory to cleanup the cache information
+
+        :param string name: repository name
+        """
+        Path.wipe(
+            os.sep.join([self.shared_zypper_dir['pkg-cache-dir'], name])
+        )
+
```

Comment:
thanks will update the code

regarding the repo I think there is not much we can do. The way zypper implemented the non-interactive flag causes the behavior you see. in interactive mode you get a question and can cancel the entire process but with non-interactive as we need it for image building the repo is just skipped for this build.